poseidon/controllers/faucet/acls.py

```diff
@@ -0,0 +1,217 @@
+                            acls_doc = yaml_in(f)
+                        else:
+                            acls_doc = yaml_in(rules_path+'/'+f)
+                        yaml_out(config_path+'/poseidon_' +
+                                 acls_filename, acls_doc)
+                        self.logger.info(
+                            'Adding {0} to config'.format(acls_filename))
+            else:
+                for f in files:
+                    if '/' in f:
+                        _, acls_filename = f.rsplit('/', 1)
+                    else:
+                        acls_filename = f
+                    if f.startswith('/'):
+                        acls_doc = yaml_in(f)
+                    else:
+                        acls_doc = yaml_in(rules_path+'/'+f)
+                    if isinstance(acls_doc, bool):
+                        self.logger.warning(
+                            'Include file {0} was not found, ACLs may not be working as expected'.format(f))
+                    else:
+                        obj_doc['include'] = ['poseidon_'+acls_filename]
+                        yaml_out(config_path+'/poseidon_' +
+                                 acls_filename, acls_doc)
+                        self.logger.info(
+                            'Adding {0} to config'.format(acls_filename))
+
+            # get defined ACL names from included files
+            for f in files:
+                acl_doc = yaml_in(f)
+                if isinstance(acl_doc, bool):
+                    self.logger.warning(
+                        'Include file {0} was not found, ACLs may not be working as expected'.format(f))
+                else:
+                    if 'acls' in acl_doc:
+                        for acl in acl_doc['acls']:
+                            acl_names.append(acl)
+
+        if 'rules' in rules_doc:
+            acls = []
+            rules = rules_doc['rules']
+
+            for rule in rules:
+                for r in rules[rule]:
+                    acls += r['rule']['acls']
+            acls = list(set(acls))
+
+            # check that acls in rules exist in the included acls file
+            if 'include' in rules_doc:
+                for acl in acls:
+                    if acl not in acl_names:
+                        self.logger.info(
+                            'Using named ACL: {0}, but it was not found in included ACL files, assuming ACL name exists in Faucet config'.format(acl))
+
+            for endpoint in endpoints:
+                port = int(endpoint.endpoint_data['port'])
+                switch = endpoint.endpoint_data['segment']
+                if switch in obj_doc['dps'] and 'acls_in' in obj_doc['dps'][switch]['interfaces'][port]:
+                    existing_acls = obj_doc['dps'][switch]['interfaces'][port]['acls_in']
+                else:
+                    existing_acls = []
+                all_rule_acls = []
+                for rule in rules:
```

Comment:
This too probably could be its own function (iterate over all rules, count matches - should make that separate function relatively easy to test too).